--- a/lib/validator.js	
+++ b/lib/validator.js	
@@ -259,9 +259,9 @@
   if (ctx.schemas[switchSchema]) {
     return {subschema: ctx.schemas[switchSchema], switchSchema: switchSchema};
   }
   // Else try walking the property pointer
-  let parsed = new URL(switchSchema,'thismessage::/');
+  let parsed = new URL(switchSchema,'thismessage:/');
   let fragment = parsed.hash;
   var document = fragment && fragment.length && switchSchema.substr(0, switchSchema.length - fragment.length);
   if (!document || !ctx.schemas[document]) {
     throw new SchemaError("no such schema <" + switchSchema + ">", schema);
